website/content/docs/secrets/databases/oracle.mdx

```diff
@@ -210,9 +232,112 @@ vault write database/config/oracle-west \
+
+   ```shell-session
+   $ vault write database/roles/my-role \
+       db_name=my-oracle-database \
+       creation_statements='CREATE USER {{username}} IDENTIFIED BY "{{password}}"; GRANT CONNECT TO {{username}}; GRANT CREATE SESSION TO {{username}};' \
+       default_ttl="1h" \
+       max_ttl="24h"
+   ```
+
+   Note: The `creation_statements` may be specified in a file and interpreted by the Vault CLI using the `@` symbol:
+
+   ```shell-session
+   $ vault write database/roles/my-role \
+       creation_statements=@creation_statements.sql \
+       ...
+   ```
+
+   See the [Commands](/vault/docs/commands#files) docs for more details.
+
+### Connect with HCP Vault using SSL
+
+When configuring the Oracle RDS options group for SSL, you must set the following:
+
+- `SQLNET.SSL_VERSION` to `1.2`
+- `SQLNET.CIPHER_SUITE` to one of `TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384`, `TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384`, `TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256`, `TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256`, `TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA`, or `TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA`
+
+If the AWS RDS Oracle instance Vault is trying to connect to uses an SSL listener, the database
+plugin will require extra configuration using the `connection_url` parameter:
+
+```shell-session
+vault write database/config/oracle \
+  plugin_name=vault-plugin-database-oracle \
+  connection_url='{{username}}/{{password}}@(DESCRIPTION=(ADDRESS=(PROTOCOL=tcps)(HOST=<host>)(PORT=<port>))(CONNECT_DATA=(SERVICE_NAME=<service_name>))(SECURITY=(SSL_SERVER_CERT_DN="<cert_dn>")(MY_WALLET_DIRECTORY=<path_to_wallet>)))' \
+  allowed_roles="my-role" \
+  username="VAULT_SUPER_USER" \
+  password="myreallysecurepassword"
+```
+
+For example, the SSL server certificate distinguished name for AWS RDS and path to the Oracle Wallet
+to use for connection and verification could be configured using:
+
+- Wallet location and permissions are managed by the HCP platform. The wallet is available at `/etc/vault.d/plugin/oracle/ssl_wallet`.
+- The distinguished name for the current AWS RDS CA is in the format `SECURITY=(SSL_SERVER_CERT_DN="C=US,ST=Washington,L=Seattle,O=Amazon.com,OU=RDS,CN=your-rds-endpoint-url")`.
+- Replace `your-rds-endpoint-url` with the endpoint for your RDS instance in the `HOST` and `DN` parameters.
+
+-> **Note:** For more information on using SSL/TLS with AWS RDS review the [Using SSL/TLS to encrypt a connetion to a DB instance](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL.html) AWS documentation.
+
+```shell-session
+vault write database/config/my-oracle-database \
+  plugin_name=vault-plugin-database-oracle \
+  connection_url="{{username}}/{{password}}@(DESCRIPTION=(ADDRESS=(PROTOCOL=tcps)(HOST=your-rds-endpoint-url)(PORT=2484))(CONNECT_DATA=(SERVICE_NAME=ORCL))(SECURITY=(SSL_SERVER_CERT_DN="C=US,ST=Washington,L=Seattle,O=Amazon.com,OU=RDS,CN=your-rds-endpoint-url")(MY_WALLET_DIRECTORY=/etc/vault.d/plugin/oracle/ssl_wallet)))" \
```

Comment:
One thing we should also add here is that 2484 only exposes it self once option group is added to the AWS RDS.

Reply:
Resolved in upcoming commit.